src/lesson/lesson.js

```diff
@@ -30,10 +30,11 @@ Lesson.prototype = {
 };
 
 /// Interface for one step of a lesson.
-function LessonStep(shortInstruction, instructions, player, initialCode,
-                    successMessage, codeSizeLimit) {
+function LessonStep(shortInstruction, instructions, commandReference,
```

Comment:
When adding a new parameter to existing API, it's generally better to put the parameter in the end and handle it as optional inside the function. In this way, existing code doesn't break. This currently breaks Lessons 1 and 2. Another option, of course, would be adding the reference to lessons 1 and 2 in this PR. But that's more work, and if you want to leave those references for later, no problem, but then we need to make sure they keep working.